packages/lesswrong/components/users/UsersProfile.jsx

```diff
@@ -32,10 +32,14 @@ const styles = theme => ({
 
 const UsersProfile = (props) => {
   if (props.loading) {
-
     return <div className="page users-profile"><Components.Loading/></div>
-
-  } else if (props.document && props.document._id && !props.document.deleted) {
+  }
+  
+  if (!props.document || !props.document._id || props.document.deleted) {
+    //eslint-disable-next-line no-console
+    console.error(`// missing user (_id/slug: ${props.documentId || props.slug})`);
+    return <Components.Error404/>
+  } else {
```

Comment:
I think our current style is to do these without an "else" clause, just have the actual return be the mainline code path.

Reply:
The next commit does that; this one only did the reorder, for diff-legibility reasons.